cfme/tests/physical_infrastructure/api/test_physical_server_api.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+import pytest
+
+from cfme.physical.provider.lenovo import LenovoProvider
+from cfme.utils import error
+from cfme.utils.rest import assert_response
+from cfme.utils.wait import wait_for
+from manageiq_client.api import APIException
+
+pytestmark = [
+    pytest.mark.tier(3),
+    pytest.mark.provider([LenovoProvider], scope='module')
+]
+
+# The max time that a single test has to perform the entire cycle of action
+# in seconds (5 min)
+TIMEOUT = 300
+
+# The time interval to perform a verification of state change
+# in seconds (1 min)
+DELAY = 60
+
+
+@pytest.fixture(scope="module")
+def physical_server(setup_provider_modscope, appliance):
+    physical_server = appliance.rest_api.collections.physical_servers[0]
+    return physical_server
+
+
+@pytest.fixture(scope="module")
+def unexistent_physical_server(setup_provider_modscope, appliance):
+    return appliance.rest_api.get_entity('physical_servers', 999999)
+
+
+def do_post_physical_server(params, appliance, physical_server):
+    """
+    Do post againist a single physical server
+    """
+    appliance.rest_api.post(physical_server.href, **params)
+    return appliance.rest_api.response.ok
+
+
+def perform_physical_server_action(action, appliance, physical_server):
+    """
+    Perform an action againist a single physical server
+    """
+    payload = {
+        "action" : action
+    }
+    return do_post_physical_server(payload, appliance, physical_server)
+
+
+def get_server_attr(attr_name, appliance, provider, physical_server):
+    provider.refresh_provider_relationships()
+    physical_server.reload()
+    return physical_server[attr_name]
```

Comment:
QUESTION: are you sure it's accessible this way? Usually it's ``physical_server.attr_name``, or ``physical_server._data[attr_name]`` when you really want to access the dictionary.
The ``physical_server.attr_name`` is preferable.

Reply:
Yeah, it is! I'm using this approach to I can pass the attribute name as parameter and reuse this code for many tests.

![image](https://user-images.githubusercontent.com/8550928/36977843-54689092-2061-11e8-8622-28d3c1d192a6.png)